src/services/local/index.js

```diff
@@ -0,0 +1,121 @@
+import Debug from 'debug';
+import errors from 'feathers-errors';
+import bcrypt from 'bcrypt';
+import passport from 'passport';
+import { Strategy } from 'passport-local';
+import { exposeConnectMiddleware } from '../../middleware';
+import { successfulLogin } from '../../middleware';
+
+const debug = Debug('feathers-authentication:local');
+const defaults = {
+  userEndpoint: '/users',
+  usernameField: 'email',
+  passwordField: 'password',
+  userProperty: passport._userProperty || 'user',
+  localEndpoint: '/auth/local',
+  tokenEndpoint: '/auth/token'
+};
+
+export class Service {
+  constructor(options = {}) {
+    this.options = options;
+  }
+
+  checkCredentials(username, password, done) {
+    const params = {
+      internal: true,
+      query: {
+        [this.options.usernameField]: username
+      }
+    };
+
+    // Look up the user
+    this.app.service(this.options.userEndpoint)
+      .find(params)
+      .then(users => {
+        // Paginated services return the array of results in the data attribute.
+        let user = users[0] || users.data && users.data[0];
+
+        // Handle bad username.
+        if (!user) {
+          return done(null, false);
+        }
+
+        return user;
+      })
+      .then(user => {
+        // Check password
+        bcrypt.compare(password, user[this.options.passwordField], function(error, result) {
```

Comment:
Yup. We could do that. I'm going to iterate on this repo a bit more over the next few days so I don't think we should delay on merging but if we get that util in commons then I'll use it here. It would actually come in handy in a few spots.